repo/blob/gdrive/file_id_cache.go

```diff
@@ -0,0 +1,130 @@
+package gdrive
+
+import (
+	"sync"
+
+	"github.com/kopia/kopia/repo/blob"
+)
+
+const (
+	changeLogCacheSize = 1 << 8
+)
+
+// FileIDCache is a cache for managing the association of blobID -> fileID.
+type FileIDCache struct {
+	// Map of blobID -> *CacheEntry.
+	Blobs sync.Map
+	// Record of recent cache changes.
+	// It's stored as a synchronized circular buffer.
+	ChangeLog [changeLogCacheSize]ChangeEntry
+	// ChangeLogIdx indicates the next location to write to.
+	// The log entry range is [ChangeLogIdx+1, ChangeLogIdx-1].
+	ChangeLogIdx int
+	// Guards access to ChangeLog.
+	ChangeLogMut sync.RWMutex
+}
+
+// CacheEntry is a blob cache entry.
+type CacheEntry struct {
```

Comment:
nit, can you make this non-exported (lowercase)?

Reply:
Done. Thanks for teaching me on go :)